_posts/2020-04-13-deprecation-warnings-rails-guide.markdown

```diff
@@ -0,0 +1,107 @@
+---
+layout: post
+title: "The Complete Guide for Deprecation Warnings in Rails"
+date: 2020-04-13 12:00:00
+categories: ["rails", "upgrades"]
+author: luciano
+---
+
+Deprecation warnings are a common thing in the software world. They are warnings that notify the developer that a specific feature (e.g. a method) will be removed soon (usually in the next [Minor](https://semver.org/) version) and should be replaced with something else.
```

Comment:
```suggestion
Deprecation warnings are a common thing in our industry. They are warnings that notify us that a specific feature (e.g. a method) will be removed soon (usually in the next [minor](https://semver.org/) or major version) and should be replaced with something else.
```

Reply:
> usually in the next [Minor] 

It depends of the kind of deprecation.

[Minor] If it has an alternative or the change is backward compatible 
[Major] the feature, method, command, whatever is removed without replacement or is not backward compatible